irohad/ametsuchi/ledger_state.hpp

```diff
@@ -6,19 +6,22 @@
 #ifndef IROHA_LEDGER_STATE_HPP
 #define IROHA_LEDGER_STATE_HPP
 
-#include <memory>
-
+#include "cryptography/hash.hpp"
 #include "interfaces/common_objects/peer.hpp"
 #include "interfaces/common_objects/types.hpp"
 
 namespace iroha {
   struct LedgerState {
-    std::shared_ptr<shared_model::interface::types::PeerList> ledger_peers;
+    shared_model::interface::types::PeerList ledger_peers;
     shared_model::interface::types::HeightType height;
+    shared_model::crypto::Hash top_hash;
 
-    LedgerState(std::shared_ptr<shared_model::interface::types::PeerList> peers,
-                shared_model::interface::types::HeightType height)
-        : ledger_peers(std::move(peers)), height(height) {}
+    LedgerState(shared_model::interface::types::PeerList peers,
+                shared_model::interface::types::HeightType height,
```

Comment:
It seems height and top_hash are bounded with each other. Maybe better to create a structure (something like `TopBlockInfo`) and pass this structure the LedgerState?